slack_wrapper.gs

```diff
@@ -95,90 +109,150 @@ class SlackEventWrapper {
     // check token
     if(!token_true){ // check that token_true has been set in script properties
       output.msg = 'error: VERIFICATION_TOKEN is not set in script properties. The command will not run. Please contact the web app developer.';
-      return(output);
+      return output;
     }  
     if (this.token !== token_true) {
       output.msg = 'error: Invalid token ' + this.token + ' . The command will not run. Please contact the web app developer.';
-      return(output);
+      return output;
     }
     
     // check request originates from our slack workspace
     if (this.teamid != teamid_true){
       output.msg = 'error: You are sending your command from an unauthorised slack workspace.';
-      return(output);
+      return output;
     }
     
     output.code=true;
-    return (output);
+    return output;
+  }
+  
+  
+  slackCmd2FctName(){
+    // match this.subtype with the function it is meant to call
+    var slackCmd2FctName = globalVariables()['SLACKCMD_TO_FUNCTIONNAME']; // this is where all the key-value pairs (event.subtype - functionName) are stored
+    if (!slackCmd2FctName.hasOwnProperty(this.subtype)){ // if no key is found for this.subtype, return error
+        return false;
+    }
+    return slackCmd2FctName[this.subtype];
   }
   
   
   checkArgSyntax(){
-    //**** todo ****//
+    // check syntax of this.args depending on function to be called
+    
+    // match doPost event.subtype with the function it is meant to call
+    var fctName = this.slackCmd2FctName();
+    if (!fctName){
+      return ('error: Sorry, the `' + this.subtype + '` command is not currently supported.');;
+    }
+    
+    // initialise output variable
+    var output = {code:false, msg:''};
+    
+    // check uniqueid syntax
+    var uniqueid_fcts = ['assign', 'volunteer', 'cancel', 'done_send_modal','done_process_modal'];
+    if (uniqueid_fcts.indexOf(fctName) > -1){ // check if fctName is expected to pass the uniqueid argument
```

Comment:
maybe move this check down into `checkArgSyntaxRegexp`? And integrate `uniqueid_fcts` into `syntax_object`

Reply:
I think this makes sense - that way all the syntax checking logic is in one place.